topics/single-cell/tutorials/scrna-ncbi-anndata/tutorial.md

```diff
@@ -0,0 +1,471 @@
+>
+> 1. {% tool [Manipulate AnnData](toolshed.g2.bx.psu.edu/repos/iuc/anndata_manipulate/anndata_manipulate/0.7.5+galaxy1) %} with the following parameters:
+>    - {% icon param-file %} *"Annotated data matrix"*: `Select all AnnData files`
+>    - *"Function to manipulate the object"*: `Transpose the data matrix, leaving observations and variables interchanged`
+>
+{: .hands_on}
+
+Now we have all the AnnData files with the data in the correct position we can combine all the data into a single AnnData file which will make it much easier to work with. This combination operation will add an additional column called **batch** which tells us which AnnData object each bit of data came from, this will be useful for further processing!
+
+> <hands-on-title>Combine AnnData objects</hands-on-title>
+>
+> 1. {% tool [Manipulate AnnData](toolshed.g2.bx.psu.edu/repos/iuc/anndata_manipulate/anndata_manipulate/0.7.5+galaxy1) %} with the following parameters:
+>    - {% icon param-file %} *"Annotated data matrix"*: `Select first Manipulate AnnData (transpose) output`
+>    - *"Function to manipulate the object"*: `Concatenate along the observations axis`
+>         - *"Annotated data matrix to add"*: `Select all other Manipulate AnnData (transpose) outputs`
+>         - *"Join method"*: `Intersection of variables`
+>         - *"Key to add the batch annotation to obs"*: `batch`
+>         - *"Separator to join the existing index names with the batch category"*: `-`
+>
+> 2. **Rename** {% icon galaxy-pencil %} output `Combined Object`
+>
+{: .hands_on}
+
+Now we have all of our gene expression data stored in a single AnnData object! Now we just need to add our metadata!
+
+# Annotating the Data
+
+The next step is to annotate our data using the information gathered from the excel sheet earlier, we will do this by leveraging the batch column generated when the objects were combined for each individual patient/sample (indicated by the batch number) we will add in the relevant annotations. First lets extract the observation data so we can manipulate it.
+
+> <hands-on-title></hands-on-title>
+>
+> 1. {% tool [Inspect AnnData](toolshed.g2.bx.psu.edu/repos/iuc/anndata_inspect/anndata_inspect/0.7.5+galaxy1) %} with the following parameters:
+>    - {% icon param-file %} *"Annotated data matrix"*: `Combined Object`
+>    - *"What to inspect?"*: `Key-indexed observations annotation (obs)`
+>
+> 2. **Rename** {% icon galaxy-pencil %} output `Combined Object`
+>
+{: .hands_on}
+
+Now for each new column we want to add we need to replace the relvant batch numbers with the appropriate metadata. After that we can cut the modified column out so it is separated. We will repeat this process for each piece of metadata we want to add, all the relevant data can be found in the first table of this tutorial.
+
+First lets add the replicate column which tells us which rows are part of pools of the same patient and tumor location.
```

Comment:
repetition of "first lets add" from previous paragraph

Reply:
I have made updates on both suggestions, thank you!